janusgraph-core/src/main/java/org/janusgraph/core/JanusGraphEdge.java

```diff
@@ -76,13 +75,10 @@ default JanusGraphVertex inVertex() {
 
     @Override
     default Iterator<Vertex> vertices(Direction direction) {
-        final List<Vertex> vertices;
         if (direction==Direction.BOTH) {
-            vertices = ImmutableList.of(vertex(Direction.OUT), vertex(Direction.IN));
-        } else {
-            vertices = ImmutableList.of(vertex(direction));
+            return Stream.of((Vertex) vertex(Direction.OUT), vertex(Direction.IN)).iterator();
         }
-        return vertices.iterator();
+        return Stream.of((Vertex) vertex(direction)).iterator();
     }
```

Comment:
We don't need immutable list here because the list isn't used anywhere and was used only for iterator creation. Thus, changed it to `Stream.of(...).iterator()`